unified-codes/apps/data-loader/src/app/v2/DataParser.ts

```diff
@@ -162,441 +156,87 @@ export class DataParser {
-        if (uc8) {
-          // and pack size code exists...
-          if (uc9) {
-            // link pack immediate to pack size.
-            if (!this.graph[uc8].children.map((child) => child.code).includes(uc9)) {
-              this.graph[uc8].children.push({ code: uc9 });
-              console.log(
-                `INFO: Linked pack immediate with code ${uc8} to pack size with code ${uc9}`
-              );
-            }
+            console.log(`INFO: Created ${item.type} node: ${JSON.stringify(node)}`);
           }
-        }
+        });
 
-        // If unit code exists...
-        if (uc7) {
-          // and pack immediate code exists...
-          if (uc8) {
-            // link unit to pack immediate.
-            if (!this.graph[uc7].children.map((child) => child.code).includes(uc8)) {
-              this.graph[uc7].children.push({ code: uc8 });
-              console.log(`INFO: Linked unit with code ${uc7} to pack immediate with code ${uc8}`);
-            }
-          }
-        }
+        const productCode = productDefinition.find(item => item.type === EEntityType.Product)?.code;
+        // If category code exists, and product code exists
+        if (uc1 && productCode) {
 
-        // If dose strength code exists...
-        if (uc6) {
-          // and unit code exists...
-          if (uc7) {
-            // link dose strength to unit.
-            if (!this.graph[uc6].children.map((child) => child.code).includes(uc7)) {
-              this.graph[uc6].children.push({ code: uc7 });
-              console.log(`INFO: Linked dose strength with code ${uc6} to unit with code ${uc7}`);
-            }
+          // link category to product.
+          if (!this.graph[uc1].children.map((child) => child.code).includes(productCode)) {
+            this.graph[uc1].children.push({ code: productCode });
+            console.log(`INFO: Linked category with code ${uc1} to product with code ${productCode}`);
           }
         }
 
-        // If dose qualifier code exists...
-        if (uc5) {
-          // and strength code exists...
-          if (uc6) {
-            // link dose qualification to strength.
-            if (!this.graph[uc5].children.map((child) => child.code).includes(uc6)) {
-              this.graph[uc5].children.push({ code: uc6 });
+        // Iterate through and assign children to parents 
+        // This starts at the top of the graph and for each node, finds the closest lower level node that has a code and name
+        // This avoids having to manually define each combination of 'levels' that are able to be connected 
+        let parentIndex = 0;
+        let childIndex = 1;
+        while (childIndex < productDefinition.length) {
+          const parent = productDefinition[parentIndex];
+          const child = productDefinition[childIndex];
+          if (child.name && child.code) {
+            if (!this.graph[parent.code].children.map((child) => child.code).includes(child.code)) {
```

Comment:
I would expect the redefinition of `child` to cause a problem here.
There's a `const child` in line 229 and now `children.map((child) =>...`
One of them will need to be renamed :)

Reply:
Thanks - I didn't notice that (interestingly they didn't seem to conflict at runtime?), but I agree it's super confusing even if it happened to work 🤣  21effa8eb